src/components/AccountManagement.vue

```diff
@@ -0,0 +1,161 @@
+<script setup lang="ts">
+import { getApp } from 'firebase/app';
+import {
+  deleteUser,
+  getAuth,
+  onAuthStateChanged,
+  signInWithEmailAndPassword,
+  updateProfile,
+  User,
+} from 'firebase/auth';
+import {
+  doc,
+  deleteDoc,
+  getDoc,
+  getFirestore,
+  setDoc,
+} from 'firebase/firestore';
+import { onBeforeUnmount, ref } from 'vue';
+
+import { router } from '../router';
+
+const auth = getAuth();
+const db = getFirestore(getApp());
+
+// Fields
+const email = ref();
+const firstName = ref();
+const lastName = ref();
+const reloginPassword = ref();
+
+// Toggles
+const editingName = ref(false);
+const showRelogin = ref(false);
+
+async function getUserInfo(user: User) {
+  if (email.value) {
+    // Already loaded user info.
+    return;
+  }
+  email.value = user.email;
+  const docSnap = await getDoc(doc(db, 'users', user.uid));
+  if (docSnap.exists()) {
+    firstName.value = docSnap.data().first_name;
+    lastName.value = docSnap.data().last_name;
+  }
+}
+
+async function deleteAccount() {
+  // If the password hasn't been entered, show the re-login form.
+  if (!reloginPassword.value) {
+    showRelogin.value = true;
+    return;
+  }
+  // If the password has been entered, log in then delete.
+  try {
+    await signInWithEmailAndPassword(auth, email.value, reloginPassword.value);
+  } catch (error) {
+    // TODO: Show errors in UI.
+    console.log(error);
+    return;
+  }
+  try {
+    await deleteDoc(doc(db, 'users', auth.currentUser!.uid));
+    await deleteUser(auth.currentUser!);
+  } catch (error) {
+    // TODO: Show errors in UI.
+    console.log(error);
+  }
+}
+
+async function updateName() {
+  if (!editingName.value) {
+    editingName.value = true;
+    return;
+  }
+  // TODO: Allow users to save if undefined e.g., no name defined initially.
+  // First, update the user profile in Firebase Auth.
+  try {
+    await updateProfile(auth.currentUser!, {
+      displayName: `${firstName.value} ${lastName.value}`,
+    });
+  } catch (error) {
+    // TODO: Show errors in UI.
+    console.log(error);
+    return;
+  }
+  // Then, also update the Firestore entry for the user.
+  try {
+    await setDoc(doc(db, 'users', auth.currentUser!.uid), {
```

Comment:
what is the difference between the displayName vs the first name/last name in the separate DB? why do we store the name in two places?

Reply:
displayName is a single field so we can't differentiate between the first and last names. We can also set a different displayName (e.g., just the first name). displayName is the only way to pass the user's name in the email templates.

As well, we're going to store the organization name and city of the user, and those must be written to the DB bc  Firebase Auth doesn't support those fields.

Reply:
Ah okay makes sense! Maybe add a comment clarifying that?